contracts/RewardPool.vy

```diff
@@ -294,89 +283,82 @@ def _claim(addr: address, ve: address, _last_token_time: uint256) -> uint256:
 
     return to_distribute
 
+
 @external
 @nonreentrant('lock')
-def claim(_addr: address = msg.sender, _lock: bool = False) -> uint256:
+def claim(user: address = msg.sender, relock: bool = False) -> uint256:
     """
-    @notice Claim fees for `_addr`
-    @dev Each call to claim look at a maximum of 50 user veCRV points.
-         For accounts with many veCRV related actions, this function
-         may need to be called more than once to claim all available
-         fees. In the `Claimed` event that fires, if `claim_epoch` is
-         less than `max_epoch`, the account may claim again.
-    @param _addr Address to claim fees for
-    @return uint256 Amount of fees claimed in the call
+    @notice Claim fees for a user
+    @dev 
+        Each call to claim looks at a maximum of 50 user veYFI points.
+        For accounts with many veYFI related actions, this function
+        may need to be called more than once to claim all available
+        fees. In the `Claimed` event that fires, if `claim_epoch` is
+        less than `max_epoch`, the account may claim again.
+    @param user account to claim the fees for
+    @param relock whether to increase the lock from the claimed fees
+    @return uint256 amount of the claimed fees
     """
-    assert not self.is_killed
-    if _lock:
-        assert _addr == msg.sender 
-
     if block.timestamp >= self.time_cursor:
         self._checkpoint_total_supply()
 
     last_token_time: uint256 = self.last_token_time
 
-    if (block.timestamp > last_token_time + TOKEN_CHECKPOINT_DEADLINE):
+    if block.timestamp > last_token_time + TOKEN_CHECKPOINT_DEADLINE:
         self._checkpoint_token()
         last_token_time = block.timestamp
 
-    last_token_time = last_token_time / TWO_WEEKS * TWO_WEEKS
+    last_token_time = last_token_time / WEEK * WEEK
 
-    amount: uint256 = self._claim(_addr, self.voting_escrow, last_token_time)
+    amount: uint256 = self._claim(user, last_token_time)
     if amount != 0:
-        token: address = self.token
-        if _lock:
-            voting_escrow: address = self.voting_escrow
-            ERC20(token).approve(voting_escrow, amount)
-            VotingEscrow(voting_escrow).deposit_for(_addr, amount)
+        # you can only relock for yourself
+        if relock and msg.sender == user or self.allowed_to_relock[user][msg.sender]:
```

Comment:
Should this be `if relock and msg.sender == user or relock and self.allowed_to_relock[user][msg.sender]:` so that we don't relock every time if we have permission?

Reply:
Wouldn't the right hand of the `and` won't be executed if relock is false?

Reply:
`or` has lower precedence than `and` -- https://docs.python.org/3/reference/expressions.html#operator-precedence -- but parentheses would make it more clear

Reply:
it's correct, but i'll add parentheses since multiple people have found the operator precedence logic non-intuitive